services/slack/intermediate.go

```diff
@@ -322,11 +323,85 @@ func buildChannelsByOriginalNameMap(intermediate *Intermediate) map[string]*Inte
 }
 
 func getNormalisedFilePath(file *SlackFile, attachmentsDir string) string {
-	filePath := path.Join(attachmentsDir, fmt.Sprintf("%s_%s", file.Id, file.Name))
-	return string(norm.NFC.Bytes([]byte(filePath)))
+	n := makeAlphaNum(file.Name, '.', '-', '_')
+	p := path.Join(attachmentsDir, fmt.Sprintf("%s_%s", file.Id, n))
+	return norm.NFC.String(p)
 }
 
-func addFileToPost(file *SlackFile, uploads map[string]*zip.File, post *IntermediatePost, attachmentsDir string) error {
+func addFileToPost(file *SlackFile, uploads map[string]*zip.File, post *IntermediatePost, attachmentsDir string, allowDownload bool) error {
+	if _, ok := uploads[file.Id]; ok || !allowDownload {
+		return addZipFileToPost(file, uploads, post, attachmentsDir)
+	}
+
+	return addDownloadToPost(file, post, attachmentsDir)
+}
+
+func addDownloadToPost(file *SlackFile, post *IntermediatePost, attachmentsDir string) error {
+	destFilePath := getNormalisedFilePath(file, attachmentsInternal)
+	log.Printf("Downloading %q to %q...\n", file.DownloadURL, destFilePath)
+
+	fullFilePath := path.Join(attachmentsDir, destFilePath)
+
+	// TODO(noxer): Use the file info to resume incomplete downloads
+	if info, err := os.Stat(fullFilePath); err == nil {
+		if info.Size() == file.Size {
+			log.Println("File has already been downloaded, skipping...")
+			return nil
+		}
+	}
+
+	destFile, err := os.Create(fullFilePath)
+	if err != nil {
+		return errors.Wrapf(err, "failed to create file %s in the attachments directory", file.Id)
+	}
+	defer destFile.Close()
+
+	resp, err := http.Get(file.DownloadURL)
```

Comment:
More of an open question, do we need to take into account any kind of auth for the download links?

Reply:
No. The URL contains a token which allows us to download the files without auth.